### PR TITLE
Copyable command when component missing in k8s publishing.

### LINKED ISSUE #933: When publishing to GKE, if Kubectl is not installed, the info on the dialog window that appears is wrong.
The command that appears on the dialog window is wrong because it says `gcloud components install Kubectl` with a capital K and it fails, it should say `gcloud components install kubectl` with lower case k.
image

### PR DESCRIPTION
**Pull request by amanda-tarafa**
Fixes #933 and #841 
- Adding a CopyablePrompt to allow commands and other texts to be copied from the prompt.
- Showing the command to install missing components when publishing to k8s in said CopyablePrompt.
- Turning the name of the missing component to lower case so the comand works as copied.

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
PTAL

File: GoogleCloudExtension/GoogleCloudExtensionUnitTests/Utils/GCloudWrapperUtilsTests.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+using Moq;
+using System;
+using System.Threading.Tasks;
+
+namespace GoogleCloudExtensionUnitTests.Utils
+{
+    [TestClass]
+    public class GCloudWrapperUtilsTests
+    {
+        private static readonly string MissingComponentTitle = Resources.GcloudMissingComponentTitle;
```
[the remark this answers is not in the document]
Comment (amanda-tarafa) [reply GoogleCloudExtension/GoogleCloudExtensionUnitTests/Utils/GCloudWrapperUtilsTests.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtensionUnitTests/Utils/GCloudWrapperUtilsTests.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                .Callback<string, string, string>((title, message, copyableMessage) =>
+                {
+                    Assert.AreEqual(MissingComponentTitle, title);
+                    Assert.AreEqual(MissingComponentMessage, message);
+                    Assert.AreEqual(MissingComponentInstallCommand, copyableMessage, false);
+                });
+
+            await GCloudWrapperUtils.VerifyGCloudDependencies(GCloudComponent.Kubectl);
+
+            _showCopyablePromptMock.Verify(a => a(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>()), Times.Once);
```
[the remark this answers is not in the document]
Comment (amanda-tarafa) [reply GoogleCloudExtension/GoogleCloudExtensionUnitTests/Utils/GCloudWrapperUtilsTests.cs]: Oh yeah, I had it like that at first, but then, if the test fails, it only says that the method wasn't called with the expected parameters, so you don't have enough info to know if it wasn't called at all or if it was called with different parameters and which are the offending parameters.
I changed it to how it is now so we get all that info.

File: GoogleCloudExtension/GoogleCloudExtension/CopyablePrompt/CopyablePromptDialogWindowContent.xaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    <theming:CommonDialogWindowBaseContent>
+        <theming:CommonDialogWindowBaseContent.Buttons>
+            <theming:DialogButtonInfo Caption="{x:Static ext:Resources.UiOkButtonCaption}" IsDefault="True" IsCancel="True"/>
+        </theming:CommonDialogWindowBaseContent.Buttons>
+
+        <StackPanel>
+            <TextBlock Text="{Binding Message}" TextWrapping="Wrap" Style="{StaticResource CommonTextStyle}"/>
+            <StackPanel Orientation="Horizontal">
+                <TextBlock Text="{Binding CopyableText}" TextWrapping="Wrap" Style="{StaticResource CommonTextStyle}" />
```
[the remark this answers is not in the document]
Comment (amanda-tarafa) [reply GoogleCloudExtension/GoogleCloudExtension/CopyablePrompt/CopyablePromptDialogWindowContent.xaml]: Done, didn't think of that.

File: GoogleCloudExtension/GoogleCloudExtensionUnitTests/Utils/GCloudWrapperUtilsTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtensionUnitTests/Utils/GCloudWrapperUtilsTests.cs]: IIRC, part of the error message of a failed verification is list of performed invocations. That would give you all of the debugging info you are looking for.

[the remark this answers is not in the document]
Comment (ILMTitan) [reply GoogleCloudExtension/GoogleCloudExtensionUnitTests/Utils/GCloudWrapperUtilsTests.cs]: From a quick test of this:
 
> Moq.MockException: 
> Expected invocation on the mock once, but was 0 times: f => "https://wrong.com"
> No setups configured.
> 
> Performed invocations: 
> DelegateInterface_Func`2_1.Invoke("https://github.com/dotnet/core/blob/master/release-notes/download-archive.md#net-core-tools-for-visual-studio-2015")

**Pull request merged**